_overviews/scala3-book/interacting-with-java.md

```diff
@@ -39,209 +39,277 @@ Note that the Java examples in this section assume that you’re using Java 11 o
 
 Here’s an example of how this works.
-Given this Java `ArrayList`:
+Given this API that returns `java.util.List[String]`:
 
+{% tabs foo-definition %}
+{% tab Java %}
 ```java
-// java
-public class JavaClass {
-  public static List<String> getStrings() {
-    return new ArrayList<String>(List.of("a", "b", "c"));
-  }
+public interface Foo {
+  static java.util.List<String> getStrings();
 }
 ```
+{% endtab %}
+{% endtabs %}
+
+You can convert that Java list to a Scala `Seq`, using the conversion utilities in the Scala `scala.jdk.CollectionConverters` object:
 
-You can convert that Java list to a Scala `Seq`, using the conversion utilities in the Scala _scala.jdk.CollectionConverters_ package:
 
+{% tabs foo-usage class=tabs-scala-version %}
+{% tab 'Scala 2' %}
+```scala
+import scala.jdk.CollectionConverters._
+import scala.collection.mutable
+
+def testList() = {
+  println("Using a Java List in Scala")
+  val javaList: java.util.List[String] = Foo.getStrings()
+  val scalaSeq: mutable.Seq[String] = javaList.asScala
+  for (s <- scalaSeq) println(s)
+}
+```
+{% endtab %}
+{% tab 'Scala 3' %}
 ```scala
-// scala
 import scala.jdk.CollectionConverters.*
+import scala.collection.mutable
 
-def testList() = 
+def testList() =
   println("Using a Java List in Scala")
-  val javaList: java.util.List[String] = JavaClass.getStrings()
-  val scalaSeq: Seq[String] = javaList.asScala.toSeq
-  scalaSeq.foreach(println)
+  val javaList: java.util.List[String] = Foo.getStrings()
+  val scalaSeq: mutable.Seq[String] = javaList.asScala
   for s <- scalaSeq do println(s)
 ```
+{% endtab %}
+{% endtabs %}
 
-Of course that code can be shortened, but the individual steps are shown here to demonstrate exactly how the conversion process works.
-
+In the above code `scalaSeq` is a wrapper over `javaList` that adapt's it to Scala's `mutable.Seq` collection.
 
 
 ## How to use Java `Optional` in Scala
 
-When you need to use the Java `Optional` class in your Scala code, import the _scala.jdk.OptionConverters_ object, and then use the `toScala` method to convert the `Optional` value to a Scala `Option`.
+When you are interacting with an API that uses the `java.util.Optional` class in your Scala code, it is fine to construct and use as in Java.
 
-To demonstrate this, here’s a Java class with two `Optional<String>` values, one containing a string and the other one empty:
+However, for idiomatic usage in Scala, such as use with `for`, you can convert it to a Scala `Option`.
 
-```java
-// java
-import java.util.Optional;
+To demonstrate this, here’s a Java API that returns an `Optional[String]` value:
 
-public class JavaClass {
-  static Optional<String> oString = Optional.of("foo");
-  static Optional<String> oEmptyString = Optional.empty();
+{% tabs bar-definition %}
+{% tab Java %}
+```java
+public interface Bar {
+  static java.util.Optional<String> optionalString();
```

Comment:
Same comment here about abstract static methods.

Reply:
I also added in an implementation.